### DISCUSSION TITLE
SORT command is always considered as Intent.WRITE

### DISCUSSION THREAD

**Discussion by SergeyKulagin** [redis/lettuce#1626] [Q&A]
Hi guys,

I was playing with Lettuce (6.1.0.M1) and Redis in a master-replica setup. 
I noticed that _Sort_ command always goes to the master node in spite of _REPLICA_PREFERRED_ configuration.
Well, I see that Sort is not considered as _ReadOnlyCommands_ which is perfectly reasonable.
But what is I'm planning to use _sort_ command a lot without writing. 
Can I somehow workaround this restriction and allow the driver to make reads from the replicas (as it was readonly command)? 

Thanks in advance,
Sergey

**Comment by mp911de** [comment]
`SORT` is categorized as write command, see

https://github.com/redis/redis/blob/efccd6353bacec9addc7278292fd8c3108d116e4/src/server.c#L779-L781

There's currently no workaround possible.

**Reply by SergeyKulagin** [reply]
Okay, thanks for the quick answer!
I'm just wandering if I use it without STORE is there any chance that it performs some write in the background and should be executed agains master only...  But anyway, the statement is clear. Thanks!

**Reply by mp911de** [reply]
Redis lately is heading to split commands and a `SORTSTORE` (as in `SUNIONSTORE`) would be probably a neat extension that would solve your issue. Check out the Redis issue tracker at https://github.com/redis/redis/.

**Reply by SergeyKulagin** [reply]
Thanks for the hint!
